main.py

```diff
@@ -80,6 +80,13 @@ def setDropdownAttr(widget, var): self.settings.settings[var] = widget.currentTe
 		def getDropdownAttr(widget, var): widget.setCurrentText(self.settings.settings[var])
 		def updateLaunchCmd(): self.invocationPreview.setPlainText(Xqemu.launchCmdToString(Xqemu.generateLaunchCmd(self.settings, True)))
 
+		def updateControllerUi():
+			# Update all four controllers
+			for i in range(1, 5):
```

Comment:
nit: not too happy about 5 appearing when we only have 4 ports. But it's a minor issue. I hope the entire codebase will get a larger refactor in the future.